pilot/pkg/model/sidecar.go

```diff
@@ -444,5 +482,23 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service) []*Se
 		}
 	}
 
-	return importedServices
+	var validServices = make(map[config.Hostname]string)
+	for _, svc := range importedServices {
+		_, f := validServices[svc.Hostname]
+		// Select a single namespace for a given hostname.
+		// If the same hostname is imported from multiple namespaces, pick the one in the configNamespace
+		// If neither are in configNamespace, an arbitrary one will be chosen
+		if !f || svc.Attributes.Namespace == configNamespace {
+			validServices[svc.Hostname] = svc.Attributes.Namespace
+		}
+	}
+
+	filteredServices := []*Service{}
+	// Filter down to just instances in scope for the service
+	for _, i := range importedServices {
+		if validServices[i.Hostname] == i.Attributes.Namespace {
+			filteredServices = append(filteredServices, i)
+		}
+	}
+	return filteredServices
```

Comment:
Just a nit. You are probably better off simply converting the imported services map to array and sort by creation time.

Reply:
You don’t have to change this if you don’t feel like.